src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    extractClusterFor(pileupToNeighbors.keys.head, pileupToNeighbors)
+  }
+
+  /**
+   * Extracts the complete cluster containing the given pileup from the map of pileups to their neighbor sets, and
+   * returns the cluster along with the reduced map of pileup to neighbor set such that no remaining pileups in the map
+   * belong in a cluster with the returned cluster.
+   * @param pileup The pileup representative of the cluster that will be returned
+   * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @return (1) The complete cluster for the given pileup; (2) Remaining map of pileup to neighbor set, all of which
+   *         are independent of the returned cluster.
+   */
+  def extractClusterFor(pileup: BreakpointPileup, pileupToNeighbors: Map[BreakpointPileup, PileupGroup]):
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    finalizeClusterFor(pileup, Seq(), pileupToNeighbors)
+  }
+
+  /**
+   * Similar to `extractClusterFor()`, but picking up from a partially complete recursive call stack.
+   * Implementation note: the implementation is not tail-recursive, but in practice the recursion depth is small.
+   * @param pileup The pileup representative of the cluster that will be returned
+   * @param existingCluster Partially complete cluster that the given pileup belongs to
+   * @param remainingNeighborGroups Additional pileup-to-neighbor-set relationships to examine and possibly add to the
+   *                                growing cluster
+   * @return (1) The complete cluster for the given pileup; (2) Remaining map of pileup to neighbor set
+   */
+  private def finalizeClusterFor(
+                          pileup: BreakpointPileup,
+                          existingCluster: PileupGroup,
+                          remainingNeighborGroups: Map[BreakpointPileup, PileupGroup]):
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    if (existingCluster.contains(pileup) && !remainingNeighborGroups.contains(pileup)) {
+      (existingCluster, remainingNeighborGroups)
+    } else {
+      val neighbors: PileupGroup = remainingNeighborGroups(pileup)
+      val reducedRemainingGroups: Map[BreakpointPileup, PileupGroup] = remainingNeighborGroups.removed(pileup)
+      if (reducedRemainingGroups.isEmpty) (neighbors :+ pileup, Map())
+      else neighbors.foldLeft[(PileupGroup, Map[BreakpointPileup, PileupGroup])](
+        (neighbors :+ pileup, reducedRemainingGroups))(
+        (currClusterAndRemaining: (PileupGroup, Map[BreakpointPileup, PileupGroup]), nextNeighbor: BreakpointPileup) => {
+          currClusterAndRemaining match {
+            case (currCluster, remaining) =>
+              finalizeClusterFor(nextNeighbor, currCluster, remaining) match {
+                case (expandedCluster, reducedRemaining) => ((currCluster ++ expandedCluster).distinct, reducedRemaining)
+              }
+          }
+        }
+      )
```

Comment:
I've been staring at this for 5 minutes and am having a really hard time following what it does.  I think the following would help me:

1. More method level doc on exactly what is returned.  I _think_ what is returned is a Map that has any `BreakpointPileup` that is added to the cluster removed as both a key in the returned `Map` and removed from all values in the map?
2. Use of more braces in the code to make it easier to see where blocks start/end
3. More inline comments on exactly what each operation is doing